lib/srv/exec.go

```diff
@@ -93,9 +94,9 @@ func NewExecRequest(ctx *ServerContext, command string) (Exec, error) {
 		}, nil
 	}
 
-	// When in recording mode, return an *remoteExec which will execute the
-	// command on a remote host. This is used by in-memory forwarding nodes.
-	if services.IsRecordAtProxy(ctx.SessionRecordingConfig.GetMode()) {
+	// If this is not a Teleport node, execute the command on a remote
+	// host. This is used by in-memory forwarding nodes.
+	if ctx.ServerSubKind == types.SubKindOpenSSHNode || services.IsRecordAtProxy(ctx.SessionRecordingConfig.GetMode()) {
```

Comment:
Comment and condition are different. `Teleport Node == (!RecordingMode && !OpenSSH)` only until a new type is added.

Reply:
Good catch, done